routing/graph.go

```diff
@@ -86,3 +91,29 @@ func (g *CachedGraph) fetchNodeFeatures(nodePub route.Vertex) (
 
 	return g.graph.FetchNodeFeatures(nodePub)
 }
+
+// FetchAmountPairCapacity determines the maximal capacity between two nodes.
+//
+// NOTE: Part of the routingGraph interface.
+func (g *CachedGraph) FetchAmountPairCapacity(nodeFrom, nodeTo route.Vertex,
+	amount lnwire.MilliSatoshi) (btcutil.Amount, error) {
+
+	// Create unified policies for all incoming connections.
+	u := newUnifiedPolicies(g.source, nodeTo, nil)
+
+	err := u.addGraphPolicies(g)
+	if err != nil {
+		return 0, err
+	}
+
+	// We may not have all policies available to describe the hop between
+	// the nodes (in the case of hop hints), which is why we return 0 in
+	// this case.
+	unifiedPolicy, ok := u.policies[nodeFrom]
+	if !ok {
+		return 0, nil
+	}
+
+	_, capacity := unifiedPolicy.getPolicy(amount, nil)
```

Comment:
Perhaps it's not ideal to have a `nil` argument here, but it seems a bit too much to initialize empty bandwidth hints for this.

Reply:
Or if `fromNode == g.source` then return 0 right at the top of this method?